### PR TITLE
owofetch: init at 0.3.1

### PR DESCRIPTION
**Pull request by nullishamy**
###### Description of changes

This adds owofetch (https://github.com/netthier/owofetch-rs) to nixpkgs! Please note this is my first package, so I'm totally unfamiliar with how the nixpkgs flow works.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by newAM — CHANGES_REQUESTED** [review]
Welcome to nixpkgs!

Two improvements:
1. Adding yourself to the maintainers list should be a separate commit (same pull-request is ok).
2. nixpkgs has a required commit message format for new packages, see [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes)

To make these changes, use these git commands (approximately):

```bash
git reset HEAD~1
git add maintainers/maintainer-list.nix
git commit -m "maintainers: add nullishamy"
git add pkgs/tools/misc/owofetch/default.nix
git add pkgs/top-level/all-packages.nix
git commit -m "owofetch: init at 0.3.1"
git push --force
```

**Comment by nullishamy** [comment]
Thanks for the quick reply! Yeah, I did read through the contributing guide but evidently missed some bits. Will push those now.

**Comment by nullishamy** [comment]
That's the rebasing ops complete.

File: pkgs/tools/misc/owofetch/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "owofetch-rs";
```
Comment (newAM) [inline pkgs/tools/misc/owofetch/default.nix:7]: The repository name is `owofetch-rs`, but in the `Cargo.toml` the package name is `owofetch`.  This also keeps it consistent with the file path.

```suggestion
  pname = "owofetch";
```

File: pkgs/tools/misc/owofetch/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  version = "0.3.1";
+
+  src = fetchFromGitHub {
+    owner = "netthier";
+    repo = pname;
```
Comment (newAM) [inline pkgs/tools/misc/owofetch/default.nix:12]:
```suggestion
    repo = "owofetch-rs";
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -37367,4 +37367,6 @@ with pkgs;
   swaysettings = callPackage ../applications/misc/swaysettings { };
 
   aitrack = libsForQt5.callPackage ../applications/misc/aitrack { };
+
+  owofetch = callPackage ../tools/misc/owofetch { };
```
Comment (newAM) [inline pkgs/top-level/all-packages.nix:37371]: Generally this list is alphabetical, not sure why it falls apart at the bottom :thinking: 

This should be moved under `owamp = callPackage ../applications/networking/owamp { };`

**Review by newAM — CHANGES_REQUESTED** [review]
Just a couple more small changes that I found :+1:

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -15484,4 +15484,10 @@
     github = "quasigod-io";
     githubId = 62124625;
   };
+  nullishamy = {
+    email = "amy.codes@null.net";
+    name = "nullishamy";
+    github = "nullishamy";
+    githubId = 99221043;
+  };
```
Comment (newAM) [inline maintainers/maintainer-list.nix:15492]: This one should be alphabetical too.

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (nullishamy) [reply pkgs/top-level/all-packages.nix:37371]: I was trying to decide if it was or not, and as noted it fell apart so i just assumed it was chronological.

**Comment by nullishamy** [comment]
Should these be squashed on to the init commit, or should it be separate?

**Comment by newAM** [comment]
> Should these be squashed on to the init commit, or should it be separate?

Squash into the relevant commits.

**Comment by nullishamy** [comment]
Should be fixed, Git is quite annoying lol.

File: pkgs/tools/misc/owofetch/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  version = "0.3.1";
+
+  src = fetchFromGitHub {
+    owner = "netthier";
+    repo = "owofetch-rs";
+    rev = "v${version}";
+    sha256 = "sha256-I8mzOUvm72KLLBumpgn9gNyx9FKvUrB4ze1iM1+OA18=";
+  };
+
+  cargoSha256 = "sha256-Rd3pMWAnNdUalDc/f1019Tiyn73JukV3K7rrXOf/INk=";
```
Comment (newAM) [inline pkgs/tools/misc/owofetch/default.nix:17]: Hm, I tried to build this and got a hash mismatch, does this build for you?

```
error: hash mismatch in fixed-output derivation '/nix/store/wny079ma72sn6cbhibsdvh24b7mmhrwc-owofetch-0.3.1-vendor.tar.gz.drv':
         specified: sha256-Rd3pMWAnNdUalDc/f1019Tiyn73JukV3K7rrXOf/INk=
            got:    sha256-rfN4QERs1H1G7ZZim//78vlxbYfU4Cx7SYYUz/QLKeU=
```

Comment (nullishamy) [reply pkgs/tools/misc/owofetch/default.nix:17]: It was building when I tested earlier, but I'm now getting the same mismatch.

Comment (nullishamy) [reply pkgs/tools/misc/owofetch/default.nix]: Pushed a hash change.

**Review by newAM — APPROVED** [review]
Looks good to me!

##### Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [ ] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches that are remotely available are fetched with `fetchpatch`

File: pkgs/tools/misc/owofetch/default.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-I8mzOUvm72KLLBumpgn9gNyx9FKvUrB4ze1iM1+OA18=";
+  };
+
+  cargoSha256 = "sha256-rfN4QERs1H1G7ZZim//78vlxbYfU4Cx7SYYUz/QLKeU=";
+
+  meta = with lib; {
+    description = "Alternative to *fetch, uwuifies all stats";
+    homepage = "https://github.com/netthier/owofetch-rs";
+    license = licenses.gpl3;
```
Comment (lourkeur) [inline pkgs/tools/misc/owofetch/default.nix:22]: If possible we should disambiguate between GPL-3 only and GPL-3 or later.

https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md

**Review by bbjubjub2494 — CHANGES_REQUESTED** [review]
Welcome aboard

File: pkgs/tools/misc/owofetch/default.nix (hunk 6 above)
Comment (nullishamy) [reply pkgs/tools/misc/owofetch/default.nix:22]: As far as I can tell, this is a `gpl3Only` project. I'm not familiar at all with the GPL licenses, could you confirm if that is the case? I'll push a fix after that.

Comment (lourkeur) [reply pkgs/tools/misc/owofetch/default.nix:22]: You are right: there is no indication either way. `only` is the more prudent interpretation then.

**Review by bbjubjub2494 — APPROVED** [review]

File: pkgs/tools/misc/owofetch/default.nix (hunk 6 above)
Comment (nullishamy) [reply pkgs/tools/misc/owofetch/default.nix:22]: Cool, I'll push that now.

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -30487,6 +30487,8 @@ with pkgs;
 
   owamp = callPackage ../applications/networking/owamp { };
 
+  owofetch = callPackage ../tools/misc/owofetch { };
```
Comment (IvarWithoutBones) [inline pkgs/top-level/all-packages.nix:30490]:
```suggestion
  owofetch = callPackage ../tools/misc/owofetch {
    inherit (darwin.apple_sdk.frameworks) Foundation DiskArbitration;
  };
```

File: pkgs/tools/misc/owofetch/default.nix [hunk 8]
```diff
@@ -0,0 +1,25 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+}:
```
Comment (IvarWithoutBones) [inline pkgs/tools/misc/owofetch/default.nix:4]:
```suggestion
{ lib
, stdenvNoCC
, rustPlatform
, fetchFromGitHub
, DiskArbitration
, Foundation
}:
```

File: pkgs/tools/misc/owofetch/default.nix [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchFromGitHub {
+    owner = "netthier";
+    repo = "owofetch-rs";
+    rev = "v${version}";
+    sha256 = "sha256-I8mzOUvm72KLLBumpgn9gNyx9FKvUrB4ze1iM1+OA18=";
+  };
+
+  cargoSha256 = "sha256-rfN4QERs1H1G7ZZim//78vlxbYfU4Cx7SYYUz/QLKeU=";
+
```
Comment (IvarWithoutBones) [inline pkgs/tools/misc/owofetch/default.nix:18]:
```suggestion
  buildInputs = lib.optionals stdenvNoCC.isDarwin [
    Foundation
    DiskArbitration
  ];
```

**Review by IvarWithoutBones — COMMENTED** [review]
This currently fails to build on darwin/MacOS, i left you some suggestions that makes it work :)

**Comment by nullishamy** [comment]
> This currently fails to build on darwin/MacOS, i left you some suggestions that makes it work :)

Thanks for that, it's all merged now, could you try rebuilding?

File: pkgs/tools/misc/owofetch/default.nix [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+, DiskArbitration
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "owofetch";
+
+  buildInputs = lib.optionals stdenvNoCC.isDarwin [
+    Foundation
+    DiskArbitration
+  ];
```
Comment (IvarWithoutBones) [inline pkgs/tools/misc/owofetch/default.nix:15]: Could you move this to below the `src`/`cargoSha256` definitions?

**Review by IvarWithoutBones — APPROVED** [review]
Looks good to me, thanks for the contribution!

Result of `nixpkgs-review pr 194070` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>owofetch</li>
  </ul>
</details>

**Pull request merged**